src/pages/setup.tsx

```diff
@@ -116,7 +116,17 @@ export default function Home() {
 
          const activeKeysets = await mint.getKeys();
 
-         addWallet(activeKeysets, url, { activeUnit: 'usd', currencies: ['usd', 'sat'] });
+         const supportsUsd = activeKeysets.keysets.some(k => k.unit === Currency.USD);
+         const supportsSat = activeKeysets.keysets.some(k => k.unit === Currency.SAT);
+
+         if (!supportsSat && !supportsUsd) {
+            throw new Error('Mint does not support USD or BTC');
+         }
+
+         addWallet(activeKeysets, url, {
+            activeUnit: supportsUsd ? Currency.USD : Currency.SAT,
+            currencies: ['usd', 'sat'],
```

Comment:
so currencies will be set to an array with both usd and set even though maybe supportsSat (or supportsUsd) is false?

Reply:
The way `addWallet` is implemented is it just tries to find keysets that match the `currencies` in the array.

So if one of the currencies isn't supported it will just be ignored in `addWallet`.

What do you think would be a better approach?